structuralcodes/codes/mc2010/_concrete_shear.py

```diff
@@ -0,0 +1,1598 @@
+"""A collection of shear formulas for concrete"""
+import warnings
+from math import pi, tan, sin, cos
+
+
+def epsilon_x(
+    E_s: float,
+    As: float,
+    Med: float,
+    Ved: float,
+    Ned: float,
+    z: float,
+    delta_e: float,
+) -> float:
+    """The maximum allowed shear resistance
+
+    fib Model Code 2010, eq. (7.3-16)
+
+    Args:
+        E (float): The E-modulus to the material in MPa
+        As (Float): The cross-section area of reinforcement in mm^2
+        Med (Float): The moment working on the material in Nmm
+        Ved (float): The shear force working on the material in N
+        Ned: (float): The normal force working on the material in N
+        z: (float): The effective shear depth in mm
+        delta_E (float): The exentricity of the load in mm
+    Returns:
+        float: The longitudinal strain"""
+    return max(
+        (
+            (1 / (2 * E_s * As))
+            * (
+                (abs(Med) / z)
+                + abs(Ved)
+                + abs(Ned) * ((1 / 2) + (delta_e / z))
+            )
+        ),
+        0,
+    )
+
+
+def v_rd(
```

Comment:
Consider making only the arguments related to Level I mandatory, and all other optional. There might be pros and cons for this approach though...